extensions/amp-ad/0.1/amp-ad.js

```diff
@@ -104,6 +104,9 @@ export class AmpAd extends AMP.BaseElement {
   }
 }
 
+// Static registery
+AmpAd.a4aRegistry = getA4ARegistery();
```

Comment:
I'm not sure that a static class property is necessary here. It shouldn't be a problem to call `getA4ARegistry` inside `upgradeComponent` and when it's needed in the tests, since it always returns the same reference. This way we don't create the `map` in a static context, which is the point of the PR

Reply:
you are right, this defeats the purpose of PR.
But here's a thing: in tests we try to modify the registry and check if the `updatedComponent` worked accordingly, hence I used a static object so that something shared could be updated.
Another way of doing this could be instantiating a class variable instead.

Reply:
I understand your reasoning, but it is already shared without being a static property of the class. The [a4aRegistry variable](https://github.com/ampproject/amphtml/pull/11257/files#diff-36b967e399c33d922bd05a23e2a4404fR56) is declared at the module-level so it should be implicitly shared whether you call `getA4ARegistry` in the `test-amp-ad.js` or `amp-ad.js` module, so making it a property of the class is unnecessary. Would you please try it that way? If that doesn't work, we can merge it like this. Thanks : )